ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivityLauncher.kt

```diff
@@ -38,7 +40,15 @@ class PaywallActivityLauncher {
      * @param offering The offering to be shown in the paywall. If null, the current offering will be shown.
      */
     @JvmOverloads
-    fun launch(offering: Offering? = null) {
-        activityResultLauncher.launch(PaywallActivityArgs(offeringId = offering?.identifier))
+    fun launch(offering: Offering? = null, fontResourceProvider: FontResourceProvider? = null) {
+        val fontMap = fontResourceProvider?.let { provider ->
+            TypographyType.values().associateBy({ it }, { provider.getFontResourceId(it) })
```

Comment:
Suggested refactor: add a constructor overload to `PaywallActivityArgs` that takes a `FontResourceProvider` and does this inside.